management_api_app/db/repositories/workspaces.py

```diff
@@ -22,10 +23,55 @@ def __init__(self, client: CosmosClient):
     def _active_workspaces_query():
         return 'SELECT * FROM c WHERE c.resourceType = "workspace" AND c.isDeleted = false'
 
-    def _get_template_version(self, template_name):
+    def _get_current_workspace_template(self, template_name) -> ResourceTemplate:
         workspace_template_repo = WorkspaceTemplateRepository(self._client)
         template = workspace_template_repo.get_current_workspace_template_by_name(template_name)
-        return template.version
+        return template
+
+    @staticmethod
+    def _convert_type_name(s: str) -> str:
+        if s == 'str':
+            return 'string'
+        if s == 'int':
+            return 'integer'
+        return 'invalid'
+
+    @staticmethod
+    def _validate_workspace_parameters(template_parameters: List[Parameter], supplied_request_parameters: dict):
+        # verify all required parameters are given
+        errors = {}
+        missing_required = []
+        required_parameters = [parameter for parameter in template_parameters if parameter.required]
```

Comment:
The templates contain some parameters that are provided down the line by the deployment service - we should ignore those -- something like this could be a solution (please test before committing)

If we don't exclude these - the user will need to provide them which we don't want

```suggestion
        system_provided = ["acr_name", "porter_driver", "tfstate_container_name", "tfstate_resource_group_name", "tfstate_storage_account_name"]
        required_parameters = [parameter for parameter in template_parameters if parameter.required and parameter.name not in system_provided]
```